synapse/config/account_validity.py

```diff
@@ -78,6 +78,8 @@ def read_config(self, config, **kwargs):
         )
 
         # Read and store template content
+        templates_config = config.get("templates") or {}
+        custom_template_directory = templates_config.get("custom_template_directory")
```

Comment:
having five copies of this code is unsatisfactory (particularly since we ought to type-check it before using it). Since `ServerConfig` is instantiated first, can we just do `self.root.server.custom_template_directory` ?

Reply:
Oh TIL the root config can be accessed like this. I agree it's a much nicer solution.